VSRAD.Syntax/SyntaxHighlighter/ErrorHighlighter/SyntaxErrorHighlighterTaggerProvider.cs

```diff
@@ -25,7 +25,7 @@ public SyntaxErrorHighlighterTaggerProvider(IDocumentFactory documentFactory)
             var document = _documentFactory.GetOrCreateDocument(buffer);
             if (document == null) return null;
 
-            return new SyntaxErrorHighlighterTagger(document.DocumentAnalysis) as ITagger<T>;
+            return new SyntaxErrorHighlighterTagger(document) as ITagger<T>;
```

Comment:
For some reason, `SyntaxErrorHighlighterTaggerProvider.CreateTagger` and `ErrorHighlighterTaggerProvider.CreateTagger` are called multiple times with the same `ITextBuffer` instance. Each time a document is open, a breakpoint placed here is hit three times. If you take a heap snapshot after opening several documents, you can see that there are more instances of `SyntaxErrorHighlighterTagger` and `ErrorHighlighterTagger` than there are of other taggers.

Replacing this line with the following solves the problem:
```
return buffer.Properties.GetOrCreateSingletonProperty(() => new SyntaxErrorHighlighterTagger(document) as ITagger<T>);
```

I couldn't pinpoint the exact conditions when `CreateTagger` is called (and the documentation does not mention this), so I think it's possible that multiple instances per buffer can be created for other taggers as well, the problem is just easier to reproduce with these two. I suggest you to use `GetOrCreateSingletonProperty` here and in other `CreateTagger` implementations.

Reply:
Thank you! I assumed that the VS under the hood handles such a situation (for example peek definition creates new View for the same Buffer) and manages resources correctly, but as it turns out, it doesn't. 

Fixed in b08ab7f